File: docs/examples/node/auth-flow-example.ts
import express from 'express';

const CORE_AUTH_BASE_URL = process.env.CORE_AUTH_BASE_URL ?? 'http://localhost:3000';
const PORT = Number(process.env.PORT ?? 4000);

const app = express();
app.use(express.json());

app.get('/login', (_req, res) => {
  const url = new URL(`${CORE_AUTH_BASE_URL}/authenticate`);
  url.searchParams.set('redirectUrl', `http://localhost:${PORT}/callback`);
  url.searchParams.set('postbackUrl', `http://localhost:${PORT}/postback`);
  url.searchParams.set('state', 'demo-state');
  res.redirect(url.toString());
});

app.post('/postback', async (req, res) => {
  const { token, state } = req.body ?? {};
  console.log('postback received', { state });

  if (!token) {
    res.status(400).json({ message: 'token missing' });
    return;
  }

  const verifyRes = await fetch(`${CORE_AUTH_BASE_URL}/verify`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ token }),
  });
  const result = await verifyRes.json();
  console.log('verification result (postback):', result);
  res.status(204).end();
});

app.get('/callback', async (req, res) => {
  const token = req.query.token?.toString();
  const state = req.query.state?.toString();

  if (!token) {
    res.status(400).send('token missing');
    return;
  }

  const verifyRes = await fetch(`${CORE_AUTH_BASE_URL}/verify`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ token }),
  });
  const result = await verifyRes.json();
  res.json({ via: 'callback', state, verification: result });
});

app.listen(PORT, () => {
  console.log(`Node auth-flow example listening on http://localhost:${PORT}`);
  console.log('Visit /login to start the flow.');
});

File: docs/examples/python/auth_flow_example.py
import os
from urllib.parse import urlencode

from flask import Flask, jsonify, redirect, request
import requests

CORE_AUTH_BASE_URL = os.environ.get("CORE_AUTH_BASE_URL", "http://localhost:3000")
PORT = int(os.environ.get("PORT", "5000"))

app = Flask(__name__)


@app.route("/login")
def login():
    params = {
        "redirectUrl": f"http://localhost:{PORT}/callback",
        "postbackUrl": f"http://localhost:{PORT}/postback",
        "state": "demo-state",
    }
    return redirect(f"{CORE_AUTH_BASE_URL}/authenticate?{urlencode(params)}")


@app.post("/postback")
def postback():
    payload = request.get_json(silent=True) or {}
    token = payload.get("token")
    state = payload.get("state")
    print("postback received", {"state": state})

    if not token:
        return (jsonify({"message": "token missing"}), 400)

    verify_res = requests.post(
        f"{CORE_AUTH_BASE_URL}/verify",
        json={"token": token},
        timeout=5,
    )
    print("verification result (postback):", verify_res.json())
    return ("", 204)


@app.route("/callback")
def callback():
    token = request.args.get("token")
    state = request.args.get("state")
    if not token:
        return ("token missing", 400)

    verify_res = requests.post(
        f"{CORE_AUTH_BASE_URL}/verify",
        json={"token": token},
        timeout=5,
    )
    return jsonify({"via": "callback", "state": state, "verification": verify_res.json()})


if __name__ == "__main__":
    app.run(port=PORT)
